### DISCUSSION TITLE
[Feature] QR Code for mobile device enrollment

### DISCUSSION THREAD

**Discussion by HALKRON** [immich-app/immich#6998] [Feature Request]
### The feature

I think it would be cool if users can enroll into immich by scanning a QR code from the web interface.
I would imagine it would be similar to device enrollment of WhatsApp or Telegram but in reverse.
This will make it easier for non-technical users to enroll their mobile devices.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
@Tyris continuing discussion here since it's a dupe - I think a PR would be welcome! Starting out simple with just setting the server URL from externalDomain is probably good.

**Reply by Tyris** [reply]
Happy to continue discussion here, but I've got a lot more detail in the other one :)

Would genuinely be interested in the appetite from the team for the "optional" suggestions I had (don't want to work on things that are unlikely to be accepted/wanted).

**Reply by bo0tzz** [reply]
I'm not entirely sure about the optional adds, hence the suggestion to start small. The ones around handling extra network settings would add awareness of that to the server and I'm not sure how I feel about that (I already wish we could do without `externalDomain`...), a sync mechanism would add entirely new complexity, and including authentication in the QR is quite common and probably sensible, but has some security implications to consider.

For now I'd suggest having the QR code just link to a server endpoint (eg `/api/server/enroll`)[^1] which then in turn can respond with a JSON document that has the full parameters. That means the QR could also be built by hand, stays quite reusable, and preemptively sidesteps any issues like the parameters being too large to reasonably encode in a QR code.

[^1]: What might also need considering is how this interacts or overlaps with the existing `/.well-known/immich` discovery endpoint.

**Reply by Tyris** [reply]
Good feedback. That wasn't the direction I was thinking with the QR codes, but gives me something to think about.

Regarding awareness of local networks/wifi etc; I understand your perpsective - but I think it makes a lot more sense for a single "administrator" (ie: the person in the house that wants everyone to use immich ;) ) to manage these globally for everyone, instead of needing to help each person configure their instance (potentially needing to explain what and why they're doing it). I think syncing would definitely be a phase 2; but think the discussion as part of this could be valuable in structing the feature.

**Reply by Tyris** [reply]
On further thought of this - my thread wasn't a duplicate of this (only optional-4 was, which I wasn't thinking would be viable as a phase 1 and was already rejected by Alex in #655 - of which this is also a duplicate).

My threads primary focus was to facilitate easy population of the server URL. This one is specifically about authentication.

Not sure where to go from here 🤷

**Reply by bo0tzz** [reply]
I don't see this request as specifically about authentication, though it could certainly include it.
